master/pkg/schemas/expconf/legacy.go

```diff
@@ -18,30 +17,23 @@ import (
 // LegacyConfig can be used to deal with configs that contain some EOL components, like searchers
 // that have been removed.
 type LegacyConfig struct {
-	checkpointStorage CheckpointStorageConfig
-	bindMounts        BindMountsConfig
-	envvars           EnvironmentVariablesMap
-	podSpec           *PodSpec
+	CheckpointStorage CheckpointStorageConfig
+	BindMounts        BindMountsConfig
+	Environment       EnvironmentConfig
+	Hyperparameters   Hyperparameters
+
+	// XXX: should we start writing code that supports not having values here?  Or put that off
```

Comment:
address?

Reply:
Actually, this comment is really a question I have that I would like you to answer, because I wasn't sure.